lib/datadog/tracing/contrib/extensions.rb

```diff
@@ -108,6 +108,46 @@ def configure(&block)
           module Settings
             InvalidIntegrationError = Class.new(StandardError)
 
+            def self.included(base)
+              base.class_eval do
+                settings :contrib do
```

Comment:
Mostly curious about the `contrib` namespace. Is the `contrib` namespace part of the specs? Also, could we add the settings directly on `lib/datadog/tracing/configuration/settings.rb` ?

Reply:
`tracing/contrib` and `tracing/configuration` are actually two separate things. We want to avoid coupling since `tracing/configuration` relates to configuration details for ALL of tracing whereas `tracing/contrib` affects only contrib related libraries. 

`peer.service` and `service_name` are related solely to integrations under the `contrib` namespace, so I think it's best if we leave those setting here. Some customers desire a core tracing package (with only tracing capabilities without contrib integrations) that doesn't include any unnecessary code. 

The code is being tested as well underneath `extensions_spec.rb` corresponding to the file

Reply:
That makes sense. 

Thanks for the explanation 😄